core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -830,7 +830,11 @@ class ReplicaManager(val config: KafkaConfig,
           val customException =
             error match {
               case Errors.INVALID_TXN_STATE => Some(error.exception("Partition was not added to the transaction"))
+              // Transaction verification can fail with a retriable error that older clients may not
+              // retry correctly. Translate these to `NOT_ENOUGH_REPLICAS`, which will cause such
+              // clients to retry the produce request.
```

Comment:
There are multiple errors that could could clients to retry produce request, we chose this one because it doesn't trigger metadata refresh.

Reply:
I've updated the comment. Let me know if it looks okay now.